pkg/process_manager_test.go

```diff
@@ -107,3 +110,73 @@ func innerStartWaitStatus(t *testing.T, pm *ProcessManager) {
 	assert.Equal(EXITED, st.State)
 	assert.Equal(1, st.ExitCode)
 }
+
+func TestStream(t *testing.T) {
+	assert := assert.New(t)
+
+	pm := new(ProcessManager)
+	defer pm.StopAll()
+
+	// Ideally this test would use an IPC mechanism for synchronization, as time
+	// is not a synchronization primitive, and relying on timing for
+	// synchronization causes tests to become unnecessarily slow and unreliable.
+
+	id, err := pm.Start("bash", "bash", "-c", `
+		for x in $(seq 1 3); do
+			sleep 0.25
+			echo "test $x"
+		done
+	`)
+	if !assert.Nil(err) {
+		return
+	}
+
+	readStream := func(postWait bool) {
+		stdout, stderr := pm.Stream(id)
+		defer func() {
+			if stdout.Close() != nil {
+				panic("unreachable")
+			}
```

Comment:
Very much a nitpick so no need to change anything here, but it's probably also okay to just discard the error when you know the implementation will never actually raise anything, e.g.
```suggestion
			// Reader implementation never returns an error
			_ = stdout.Close()
```

Panicking like this is okay too, just verbose.

Reply:
Yeah, reasonable. I'm just used to the Rust ecosystem where using `Result::expect` is typically considered best practice and is basically a less verbose version of the if + panic.

Reply:
Ignoring the error entirely in this kind of case is probably better for perf, so maybe best case would be conditional compilation to panic in debug mode and not check anything in release mode.